salt/_states/metalk8s_cordon.py

```diff
@@ -70,4 +69,19 @@ def node_uncordoned(name, **kwargs):
     name
         the name of the node
     '''
+    all_node_taints = __salt__['metalk8s_kubernetes.node_taints'](
+        name, **kwargs
+    )
+    key_to_remove = "node.kubernetes.io/unschedulable"
+
+    new_taints = [
+        t for t in all_node_taints
+        if t['key'] not in key_to_remove
```

Comment:
You need to use a list for `key_to_remove`
Here you will remove all taint that have a key who contains a part of  "node.kubernetes.io/unschedulable" so for example you will remove taint "kubernetes.io/unsche" (that does not exists btw but we should not do that).

OR much better you should change this line
```suggestion
        if t['key'] == key_to_remove
```

Reply:
Yes.. makes sense here to be specific on the taint we want to remove